src/ui/components/search/filtersearchcomponent.js

```diff
@@ -102,6 +96,21 @@ export default class FilterSearchComponent extends Component {
       this.search();
     });
 
+    /**
+     * The filter string to use for the provided query
+     * Optionally provided
+     * @type {string}
+     */
+    this.filter = config.filter || this.core.globalStorage.getState(`${StorageKeys.FILTER}.${this.name}`) || Filter.empty();
+    if (typeof this.filter === 'string') {
+      try {
+        this.filter = JSON.parse(this.filter);
+      } catch (e) {}
+    }
+
+    const filterNode = this._buildFilterNode(this.query, this.filter);
+    this.core.setStaticFilterNodes(this.name, filterNode);
```

Comment:
FilterSearch didn't call this.core.setFilter in the constructor. Was that a bug? Someone could have specified a config.filter and it wouldn't have been added to globalStorage.

Reply:
Yes I believe this was a bug. core.setFilter was being called in the onCreate of the filter for a long time, but was removed in #196 . (see https://github.com/yext/answers/pull/196/commits/2df0026140e80f2ca4eacedada71597823bfe490) . This PR looks like it was done purely for refactoring purposes, and should not have changed functionality, but it looks like this was not the case. 

There is also reference to a this._saveQueryAndFilter that straight up does not exist...? Even going back as far as to when lines referencing this._saveQueryAndFilter were created the function _saveQueryAndFilter() does not exist so I am still quite confused as to what was happening here. Either way it probably doesn't concern this PR.